### ISSUE TITLE
@member_count and @subcollection_count should be presenter methods defaulting to 0

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#2673]
### Descriptive summary

There are two attributes referenced directly in the dashboard collection show page and possible public show page as well.  These should be accessed through methods in the presenter.  At least one of them is nil for non-nestable collections, which causes an exception to be thrown.

### Rationale

All code should be encapsulated in the presenter class instead of being in the view.

### Expected behavior

When viewing the show page for a non-nestable collection, it should display the show page without errors.

### Actual behavior

When viewing the show page for a non-nestable collection, it throws exception `ActionView::Template::Error: undefined method `>' for nil:NilClass ` at line (views/hyrax/dashboard/collections/show.html.erb#L42)[https://github.com/samvera/hyrax/blob/master/app/views/hyrax/dashboard/collections/show.html.erb#L42] which checks whether there are any members or subcollections.

### Steps to reproduce the behavior

1. Prereq: As admin, from Dashboard -> Settings -> Collection Types, create a collection type that has Nestable unchecked.
1. From Dashboard -> Collections, create a collection of the non-nestable type
1. From Dashboard -> Collections, click the title of the new collection

Verify that the page displays correctly.

1. Click Public View of Collection

Verify that the page displays correctly.

**Comment by elrayle** [comment]
I did a quick fix for this in my local hyrax as a stop gap.  A partial PR along these lines would be accepted to stop the exception, followed by another PR which moves the appropriate code to the presenter.

In app/controllers/hyrax/dashboard/collections_controller.rb, add the initialization line for subcollection_count...
```
        def query_collection_members
          member_works
          @subcollection_count = 0 # initialize since member_subcollections won't be called if not nestable
          member_subcollections if collection.collection_type.nestable?
          parent_collections if collection.collection_type.nestable? && action_name == 'show'
        end
```

**Comment by elrayle** [comment]
PR #2688 fixed the bug that caused non-nested collections to get an exception when viewing the show page

Remaining work:
* Not all occurrences of `@subcollection_count` were replaced with `@presenter.subcollection_count`.  Remaining
* Other instance variables are also accessed directly in collection show views (e.g. @members_count)

Remaining work is addressed by issue #2698.
